faunadb/query.py

```diff
@@ -286,89 +268,130 @@ def date(string):
   See also :py:func:`select_with_default`."""
-  return {"select": path, "from": from_}
+  return _q({"select": path, "from": from_})
 
 
 def select_with_default(path, from_, default):
   """See the `docs <https://faunadb.com/documentation/queries#misc_functions>`__."""
-  return {"select": path, "from": from_, "default": default}
+  return _q({"select": path, "from": from_, "default": default})
 
 
 def add(*numbers):
   """See the `docs <https://faunadb.com/documentation/queries#misc_functions>`__."""
-  return {"add": _varargs(numbers)}
+  return _q({"add": _varargs(numbers)})
 
 
 def multiply(*numbers):
   """See the `docs <https://faunadb.com/documentation/queries#misc_functions>`__."""
-  return {"multiply": _varargs(numbers)}
+  return _q({"multiply": _varargs(numbers)})
 
 
 def subtract(*numbers):
   """See the `docs <https://faunadb.com/documentation/queries#misc_functions>`__."""
-  return {"subtract": _varargs(numbers)}
+  return _q({"subtract": _varargs(numbers)})
 
 
 def divide(*numbers):
   """See the `docs <https://faunadb.com/documentation/queries#misc_functions>`__."""
-  return {"divide": _varargs(numbers)}
+  return _q({"divide": _varargs(numbers)})
 
 
 def modulo(*numbers):
   """See the `docs <https://faunadb.com/documentation/queries#misc_functions>`__."""
-  return {"modulo": _varargs(numbers)}
+  return _q({"modulo": _varargs(numbers)})
 
 
 def lt(*values):
   """See the `docs <https://faunadb.com/documentation/queries#misc_functions>`__."""
-  return {"lt": _varargs(values)}
+  return _q({"lt": _varargs(values)})
 
 
 def lte(*values):
   """See the `docs <https://faunadb.com/documentation/queries#misc_functions>`__."""
-  return {"lte": _varargs(values)}
+  return _q({"lte": _varargs(values)})
 
 
 def gt(*values):
   """See the `docs <https://faunadb.com/documentation/queries#misc_functions>`__."""
-  return {"gt": _varargs(values)}
+  return _q({"gt": _varargs(values)})
 
 
 def gte(*values):
   """See the `docs <https://faunadb.com/documentation/queries#misc_functions>`__."""
-  return {"gte": _varargs(values)}
+  return _q({"gte": _varargs(values)})
 
 
 def and_expr(*booleans):
   """See the `docs <https://faunadb.com/documentation/queries#misc_functions>`__."""
-  return {"and": _varargs(booleans)}
+  return _q({"and": _varargs(booleans)})
 
 
 def or_expr(*booleans):
   """See the `docs <https://faunadb.com/documentation/queries#misc_functions>`__."""
-  return {"or": _varargs(booleans)}
+  return _q({"or": _varargs(booleans)})
 
 
 def not_expr(boolean):
   """See the `docs <https://faunadb.com/documentation/queries#misc_functions>`__."""
-  return {"not": boolean}
+  return _q({"not": boolean})
 
 #endregion
 
 #region Helpers
 
+class _Expr(object):
+  """
+  Used to mark values that have been converted to queries.
+  :any:`to_query` simply unwraps these.
+  """
+
+  def __init__(self, value):
+    self.value = value
+
+
+def to_query(value):
```

Comment:
does this need to be documented? I'm not sure how helpful/relevant it is to go into the implementation details of how expression submission works, rather than saying things like "dicts are evaluated as objects" or something.